src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -305,6 +336,23 @@ def flexible_devices(devices)
       def reusable_by_md(planned_devices)
         planned_devices.select { |d| d.respond_to?(:raid_name) }
       end
+
+      # Returns a list of planned partitions adjusting the size
+      #
+      # All partitions which sizes are specified as percentage will get their minimal and maximal
+      # sizes adjusted.
+      #
+      # @param planned_partitions [Array<Planned::Partition>] List of planned partitions
+      # @return [Array<Planned::Partition>] New list of planned partitions with adjusted sizes
+      def sized_partitions(planned_partitions)
+        planned_partitions.map do |part|
+          new_part = part.clone
+          next new_part unless new_part.percent_size
+          disk = original_graph.find_by_name(part.disk)
+          new_part.max = new_part.min = new_part.size_in(disk)
```

Comment:
I'm just wondering (I would really need to check)... since `#size_in` takes no alignment and such things into account... Will this work as expected? I mean, what happens if we set `min` and `max` to the same non-aligned value?

And in a similar fashion, can it happen that a drive with two partitions defined as 50% each ends up with unused space due to rounding?

I know these questions are probably not new, but something that was already present (and properly solved) in the algorithms since the beginning. But every time I see this I cannot prevent my brain to start spinning.

Reply:
As discussed in a call, @imobachgs will create a separate PBI about checking and/or unit-testing several situations regarding grain and alignment in AutoYaST like:

- Two partitions/LVs with size 50% when the capacity of the disk/VG is uneven (let's say 101 extents available).
- What happens if we set min and max to the same value which is not divisible by the grain
- What happens if we set min and max to the same value that is smaller than the grain
- Whether the final 16 KiB in a GPT can break some AutoYaST profile somehow

Our expectations is that all those situations are properly handled by the following steps (the ones that create the real devices in the devicegraph based on the Planned Devices generated by this part), but double checking and having unit tests will be nice.